blocks/media-upload-button/index.js

```diff
@@ -78,11 +92,19 @@ class MediaUploadButton extends Component {
 		}
 
 		if ( gallery ) {
```

Comment:
In this case, should we expect the `selection` object to be dependent on the `multiple` prop instead of the `gallery` prop. I'm guessing the selection shape depend on whether the selection is multiple or not, and not whether it's a gallery view or not?